components/usage/pkg/apiv1/usage.go

```diff
@@ -121,6 +122,116 @@ func (s *UsageService) ListBilledUsage(ctx context.Context, in *v1.ListBilledUsa
 	}, nil
 }
 
+func (s *UsageService) ListUsage(ctx context.Context, in *v1.ListUsageRequest) (*v1.ListUsageResponse, error) {
+	to := time.Now()
+	if in.To != nil {
+		to = in.To.AsTime()
+	}
+	from := to.Add(-maxQuerySize)
+	if in.From != nil {
+		from = in.From.AsTime()
+	}
+
+	if from.After(to) {
+		return nil, status.Errorf(codes.InvalidArgument, "Specified From timestamp is after To. Please ensure From is always before To")
+	}
+
+	if to.Sub(from) > maxQuerySize {
+		return nil, status.Errorf(codes.InvalidArgument, "Maximum range exceeded. Range specified can be at most %s", maxQuerySize.String())
+	}
+
+	if in.Pagination.PerPage < 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "Number of items perPage needs to be positive (was %d).", in.Pagination.PerPage)
+	}
+
+	if in.Pagination.Page < 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "Page number needs to be 0 or greater (was %d).", in.Pagination.Page)
+	}
+
+	attributionId, err := db.ParseAttributionID(in.AttributionId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "AttributionID '%s' couldn't be parsed (error: %s).", in.AttributionId, err)
+	}
+
+	order := db.DescendingOrder
+	if in.Order == v1.ListUsageRequest_ORDERING_ASCENDING {
+		order = db.AscendingOrder
+	}
+
+	var perPage int64 = 50
+	if in.Pagination.PerPage > 0 {
+		perPage = in.Pagination.PerPage
+	}
+	var page int64 = in.Pagination.Page
+	var offset int64 = perPage * page
+
+	listUsageResult, err := db.FindUsage(ctx, s.conn, &db.FindUsageParams{
+		AttributionId: db.AttributionID(in.GetAttributionId()),
```

Comment:
```suggestion
		AttributionId: attributionId,
```

Reply:
Also, in Go it's a common style to capitalize shorthands. In this case, `ID` instead of `Id`